helm/templates/promtail/deployment.yaml

```diff
@@ -0,0 +1,91 @@
+apiVersion: extensions/v1beta1
+kind: DaemonSet
+metadata:
+  name: {{ template "promtail.fullname" . }}
+  labels:
+    app: {{ template "promtail.name" . }}
+    chart: {{ template "promtail.chart" . }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+{{- with .Values.promtail.annotations }}
+  annotations:
+{{ toYaml . | indent 4 }}
+{{- end }}
+spec:
+  replicas: {{ .Values.promtail.replicas }}
+  selector:
+    matchLabels:
+      app: {{ template "promtail.name" . }}
+      release: {{ .Release.Name }}
+  strategy:
+    type: {{ .Values.promtail.deploymentStrategy }}
+  {{- if ne .Values.promtail.deploymentStrategy "RollingUpdate" }}
+    rollingUpdate: null
+  {{- end }}
+  template:
+    metadata:
+      labels:
+        app: {{ template "promtail.name" . }}
+        release: {{ .Release.Name }}
+{{- with .Values.promtail.podAnnotations }}
+      annotations:
+{{ toYaml . | indent 8 }}
+{{- end }}
+    spec:
+      serviceAccountName: {{ template "promtail.serviceAccountName" . }}
+{{- if .Values.promtail.priorityClassName }}
+      priorityClassName: {{ .Values.promtail.priorityClassName }}
+{{- end }}
+      containers:
+        - name: {{ .Chart.Name }}
+          image: "{{ .Values.promtail.image.repository }}:{{ .Values.promtail.image.tag }}"
+          imagePullPolicy: {{ .Values.promtail.image.pullPolicy }}
+          args: 
+            - "-config.file=/etc/promtail/promtail.yaml"
+            - "-client.url=http://loki:3100/api/prom/push"
```

Comment:
This should probably be `{{ template "loki.fullname" . }}` (the name of the service), no?

Reply:
And `{{ .Values.loki.service.port }}`